front/src/applications/operationalStudies/hooks/useSetupItineraryForTrainUpdate.ts

```diff
@@ -88,20 +85,67 @@ export const useSetupItineraryForTrainUpdate = (
               const stepsCoordinates = pathfindingResult.path_items_positions.map((position) =>
                 getPointCoordinates(geometry, pathfindingResult.length, position)
               );
-
-              const formatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => ({
-                ...step,
-                coordinates: stepsCoordinates[i],
-                positionOnPath: pathfindingResult.path_items_positions[i],
-              }));
-
               const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
                 operational_points,
                 geometry,
                 pathfindingResult.length
               );
 
-              const allWaypoints = upsertViasInOPs(suggestedOperationalPoints, formatedPathSteps);
+              const updatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => {
+                const correspondingOp = suggestedOperationalPoints.find(
+                  (suggestedOp) =>
+                    'uic' in step &&
+                    suggestedOp.uic === step.uic &&
+                    suggestedOp.ch === step.secondary_code
+                );
+
+                const correspondingSchedule = trainSchedule.schedule?.find(
+                  (schedule) => schedule.at === step.id
+                );
+
+                const { kp, name, ch } = correspondingOp || {};
+
+                const {
+                  arrival,
+                  stop_for: stopFor,
+                  locked,
+                  on_stop_signal: onStopSignal,
+                } = correspondingSchedule || {};
+
+                const stepWithoutSecondaryCode =
+                  'secondary_code' in step ? omit(step, ['secondary_code']) : step;
+
+                return {
+                  ...stepWithoutSecondaryCode,
+                  ch,
+                  kp,
+                  name,
+                  positionOnPath: pathfindingResult.path_items_positions[i],
+                  arrival,
```

Comment:
For arrival, we get an ISO8601 format (ex: "PT60M" > arrival needs to be 60 minutes after the departure). So as we want to display the arrival, in the timestops table, as a time (ex: "11:35:00"), we need to be able to add the start time and the arrival and format it in timestring.

To do that, you can create a new util (that will be very handy for other computations) : 
`
/**
*Add up a ISO8601 date with another one in local time
*/
export const combineIsoDates(startTime: string, duration: string) => dayjs(startTime)
            .add(ISO8601Duration2sec(duration), 'second')
            .format()
`

```suggestion
                  arrival: arrival ? combineIsoDates(start_time, arrival) : arrival,
```

Reply:
Suggestion for alternative naming: `addDurationToIsoDate`